jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/SaveQueryDialog.kt

```diff
@@ -54,31 +57,54 @@ class SaveQueryDialog(
     override fun doValidate(): ValidationInfo? = validateQueryName(view)
     override fun getOKAction(): Action = action
 
-    fun checkQueryName(queryName: String): Boolean {
-        val request = DescribeQueryDefinitionsRequest.builder().queryDefinitionNamePrefix(queryName).build()
-        val response = client.describeQueryDefinitions(request)
-        return response.queryDefinitions().isEmpty()
+    private suspend fun getExistingQueryId(queryName: String): String? {
+        val definitions = withTimeout(AwsResourceCache.DEFAULT_TIMEOUT.toMillis()) {
```

Comment:
Why are we doing this `withTimeout` if the resource cache times out and throws already? won't this potentially throw two different types of exceptions?

Reply:
getResource is the version that doesn't timeout so i don't think that's an issue. the resource cache isn't very ergonomic with coroutines though and i'm not sure what the best way to fix it is